test/testBody.cpp

```diff
@@ -1,96 +1,151 @@
-#include "gtest/gtest.h"
 #include "RequestParser.hpp"
+#include "gtest/gtest.h"
 
 // VALID BODY TEST SUITE
 
-TEST(RequestParser_ValidBody, ChunkedBody) {
-	RequestParser	p;
-	HTTPRequest		request;
+TEST(RequestParser_ValidBody, ChunkedBody)
+{
+	RequestParser p;
+	HTTPRequest request;
 
-	p.parseHttpRequest("GET /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: www.example.com\r\nTransfer-Encoding: gzip, chunked\r\n\r\n6\r\nhello \r\n6\r\nworld!\r\n0\r\n\r\n", request);
+	p.parseHeader("POST /search?query=openai&year=2024#conclusion HTTP/1.1\r\nHost: "
```

Comment:
I think right now you have to call parseHeader() before parseBody() since state is saved in requestParser. 
If moved into struct no need for that

Reply:
Well, I do it because I want to determine whether the body is chunked or not. And also to retrieve the content-length. 

I could also just set it in the HTTPRequest and not parse the header. If need be, I can change.

Reply:
Test are only calling parseBody() now. The required conditions are set in the test fixture.